### DISCUSSION TITLE
Accessing variables from conf.py in .md files (e.g. release)

### DISCUSSION THREAD

**Discussion by njr0** [executablebooks/meta#868] [Q&A] [answered]
I've been using MyST with Sphinx for a few weeks now, and it's fantastic.

Up to now, I've been using an edited version of `index.rst` and including Markdown files for all the main content, but today I switched to using `index.md`. I was able to move everything across except that I can't see how to access the release number in the markdown.

Previously, I use

    from project.version import __version__

    release = __version__

in`conf.py` to get the version number of the code and then used

    Version |release|

in `index.md` and other places to set the version number there. But I can't see how to do that using MyST.
If I used `|release|` it just prints |release|.

Is there a way to access variables like `release` in MyST?

**Comment by NickleDave** [comment, marked as answer]
Not sure if there's a way to use the RST substitions directly, but I think you can enable the MyST "substitions" extension to get similar behavior:  
https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#substitutions-with-jinja2

**Reply by njr0** [reply]
Indeed, this works perfectly. I had actually thought about that, but somehow failed to try it.

In case it helps anyone, specifically I

 * Added `'substitution'` to `myst_enable_extensions`
 * Used:

        myst_substitutions = {
            'version': str(__version__)
        }
 
* Used

        Version {{ version }}

in the Markdown, and this worked perfectly.

**Comment by choldgraf** [comment]
I think that you have access to a `sub-ref` directive that lets you mimic the substitution reference behavior: https://myst-parser.readthedocs.io/en/latest/syntax/roles-and-directives.html#myst-only

So you can do things like `` {sub-ref}`today` ``

I also believe that @NickleDave's suggestion above would work too!

**Reply by njr0** [reply]
And this also worked perfectly, Chris:

    Version {sub-ref}`release`

Many thanks!
